market-generator/src/main/java/com/market/banica/generator/service/task/TickTimerTask.java

```diff
@@ -31,9 +35,9 @@ public void run() {
                 generateRandomPrice(),
                 new Date().getTime());
 
-        tickGenerator.getGeneratedTicks().add(marketTick);
+        marketState.addTickToMarketState(marketTick);
 
-        TickTimerTask nextTick = new TickTimerTask(tickGenerator, good);
+        TickTimerTask nextTick = new TickTimerTask(tickGenerator, good, marketState);
         tickGenerator.getTickTimerTasks().put(good.getName(), nextTick);
```

Comment:
it will be better if this code is wrapped in TickGenerator class and tickTimerTasks and TickTimer are visible only for the TickGenerator, so the other classes don't need to know how the task is scheduled, but just to pass the properties needed to schedule the task

Reply:
Updated in another pull request https://github.com/banica-org/banica-with-pumpkin/pull/44